pandas/core/indexes/interval.py

```diff
@@ -224,7 +226,14 @@ class IntervalIndex(IntervalMixin, Index, accessor.PandasDelegate):
     # Immutable, so we are able to cache computations like isna in '_mask'
     _mask = None
 
-    _raw_inherit = {"_ndarray_values", "__array__", "overlaps", "contains"}
+    _raw_inherit = {
+        "_ndarray_values",
+        "__array__",
+        "overlaps",
+        "contains",
+        "__eq__",
+        "__ne__",
```

Comment:
to be consistent with our other EAs, these need to be dispatched/wrapped the same way they are in datetimelike or categorical.  im planning to move the relevant code to indexes.extension so this can re-use the existing code.

The relevant test will be `arr == series` --> Series, `idx == series` --> ndarray[bool]

Reply:
You should now be able to use indexes.extension.make_wrapped_comparison_op